src/mon/MgrMonitor.cc

```diff
@@ -70,63 +70,33 @@ static ostream& _prefix(std::ostream *_dout, Monitor &mon,
 
 // the system treats always_on_modules as if they provide built-in functionality
 // by ensuring that they are always enabled.
-const static std::map<uint32_t, std::set<std::string>> always_on_modules = {
-  {
-    CEPH_RELEASE_OCTOPUS, {
-      "crash",
-      "status",
-      "progress",
-      "balancer",
-      "devicehealth",
-      "orchestrator",
-      "rbd_support",
-      "volumes",
-      "pg_autoscaler",
-      "telemetry",
-    }
-  },
-  {
-    CEPH_RELEASE_PACIFIC, {
-      "crash",
-      "status",
-      "progress",
-      "balancer",
-      "devicehealth",
-      "orchestrator",
-      "rbd_support",
-      "volumes",
-      "pg_autoscaler",
-      "telemetry",
-    }
-  },
-  {
-    CEPH_RELEASE_QUINCY, {
-      "crash",
-      "status",
-      "progress",
-      "balancer",
-      "devicehealth",
-      "orchestrator",
-      "rbd_support",
-      "volumes",
-      "pg_autoscaler",
-      "telemetry",
-    }
-  },
-  {
-    CEPH_RELEASE_REEF, {
+static const std::map<uint32_t, std::set<std::string>>& release_always_on_modules() {
+  static std::map<uint32_t, std::set<std::string>> always_on_modules;
+  static std::once_flag did_init;
+  std::call_once(did_init, []() {
```

Comment:
What is the reason for introducing `std::call_once()` complexity?  Wouldn't a regular function-local static work and be more efficient?  Something like:
```
  static std::set<std::string> always_on_modules = {
    "crash",
    "status",
    "progress",
    "balancer",
    "devicehealth",
    "orchestrator",
#ifdef WITH_RBD
    "rbd_support",
#endif
#ifdef WITH_CEPHFS
    "volumes",
#endif
    "pg_autoscaler",
    "telemetry",
  };
  static std::map<uint32_t, std::set<std::string>> always_on_modules_map = {
    {CEPH_RELEASE_OCTOPUS, always_on_modules},
    {CEPH_RELEASE_PACIFIC, always_on_modules},
    {CEPH_RELEASE_QUINCY, always_on_modules},
    {CEPH_RELEASE_REEF, always_on_modules},
  };
  return always_on_modules_map;
```

Reply:
Well, this approach is indeed simpler. However, I'd prefer it with my current naming convention of `octopus_modules`, because if the next release expands on the list it won't be able to reuse the same set and would need to come up with another name or encode it in place.

Reply:
anyway, I like this option, will pick the suggestion